src/DocumentFragment.js

```diff
@@ -0,0 +1,71 @@
+  // ### DocumentFragment Shim
+  //
+  // A representation of a document fragment that can be normally generated
+  // by calling `document.createDocumentFragment`. Since the document fragment
+  // is itself a node, it inherits all properties of a node.
+module.exports = (function(){
+  "use strict";
+
+  var Node = require('./Node');
+  var voidHtmlElements = ["area", "base", "br", "col", "command", "embed", "hr",
+                          "img", "input", "keygen", "link", "meta", "param",
+                          "source", "track", "wbr"];
+
+  function assertExists(name, value) {
+    if(!value || value === "") {
```

Comment:
Be extremely careful here, because this code does not do what you claim it does based on the function. The *only* real test you should be doing here is `if (typeof value === "undefined") { throw }`, since values like `false`, 0 and `""` are all falsy, while being perfectly valid attribute values.